fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/impl/SearchProviderSpec.groovy

```diff
@@ -526,6 +526,31 @@ abstract class SearchProviderSpec<T extends SearchProvider> extends Specificatio
         resultsPage.numResults == 3
     }
 
+    def "An empty string filter value should return all results"() {
+        setup: "Given a filter that will match every row"
+        /* Expected rows, not necessarily in this order:
+                name: "owl", description: "this is an owl"
+                name: "hawk", description: "this is a raptor"
+                name: "eagle", description: "this is a raptor"
+                name: "kumquat", description: "this is not an animal"
+        */
+        Set<ApiFilter> filters = [
+                buildFilter("animal|desc-startswith[this]"),
+                buildFilter("animal|desc-contains[\"\"]")
+        ]
+        and: "We get the second page, where each page has two rows (so both pages should have two results)"
```

Comment:
`We get the second page, where each page has two rows` sounds good enough already